From 7c60720f60ea57ec5beede81c236c7afcd32562a Mon Sep 17 00:00:00 2001
From: Chandrakala Chavva <cchavva@caviumnetworks.com>
Date: Thu, 8 Aug 2013 15:27:43 -0700
Subject: [PATCH 264/974] MIPS: OCTEON: Use correct L2C CSR for cache locking.

Read fuse register to determine L2C locking for TLB and interrupt handler.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
[Original patch taken from Cavium SDK 3.1.2-568]
Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
---
 arch/mips/cavium-octeon/setup.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/mips/cavium-octeon/setup.c b/arch/mips/cavium-octeon/setup.c
index f9ae64b..6fc302a 100644
--- a/arch/mips/cavium-octeon/setup.c
+++ b/arch/mips/cavium-octeon/setup.c
@@ -840,10 +840,11 @@ append_arg:
 #ifdef CONFIG_CAVIUM_OCTEON_LOCK_L2
 static int __init octeon_l2_cache_lock(void)
 {
-	int is_octeon2 = (current_cpu_type() == CPU_CAVIUM_OCTEON2);
+	bool is_octeon = !(current_cpu_type() == CPU_CAVIUM_OCTEON2 ||
+			   current_cpu_type() == CPU_CAVIUM_OCTEON3);
 
-	if ((is_octeon2 && (cvmx_read_csr(CVMX_MIO_FUS_DAT3) & (3ull << 32)))
-	    || (!is_octeon2 && (cvmx_read_csr(CVMX_L2D_FUS3) & (3ull << 34)))) {
+	if ((!is_octeon && (cvmx_read_csr(CVMX_MIO_FUS_DAT3) & (3ull << 32)))
+	    || (is_octeon && (cvmx_read_csr(CVMX_L2D_FUS3) & (3ull << 34)))) {
 		pr_info("Skipping L2 locking due to reduced L2 cache size\n");
 	} else {
 		u32 __maybe_unused my_ebase = read_c0_ebase() & 0x3ffff000;
-- 
2.6.2

